lib/AbstractHandle/Utils/MongoUtil.py

```diff
@@ -109,6 +107,10 @@ def get_hid_counter(self):
         else:
             return 0
 
+    def increase_counter(self):
+        self._inc_counter()
```

Comment:
This still has a race condition since you increment then fetch. Another increment could happen in that time, in which case the ids colllide. This operation has to be atomic.

Reply:
👍 I see what you mean now. Thanks.